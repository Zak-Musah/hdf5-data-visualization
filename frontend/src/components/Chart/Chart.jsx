import React, { useEffect, useState } from "react";
import Plot from "react-plotly.js";

import { chartStyling, Labels } from "../Helpers/Constants";

function Chart({ fileData }) {
  const [dataToPlot, setDataTop] = useState([]);

  useEffect(() => {
    if (Object.keys(fileData).length > 0) {
      const data = fileData;
      const time = data.time;
      const glucose = data.glucose;
      const measurement = data.measurement;
      const chartsObject = [
        { xaxis: time, yaxais: glucose, title: "Glucose Dataset" },
        { xaxis: time, yaxais: measurement, title: "Measurement Dataset" },
      ];
      setDataTop(chartsObject);
    }
  }, [fileData]);

  return (
    <div className="row">
      {dataToPlot.length > 0 &&
        Labels &&
        dataToPlot.map((obj, i) => (
          <div className="col-12">
            <Plot
              data={[
                {
                  x: obj.xaxis,
                  y: obj.yaxais,
                  type: "date",
                  mode: "line",

                  line: {
                    width: 3,
                    color: "#e6347d",
                  },
                  hoverinfo: "closest",
                },
              ]}
              layout={{
                autosize: true,
                showlegend: false,
                showgrid: true,
                gridcolor: chartStyling.gridcolor,
                textColor: chartStyling.textColor,
                title: obj.title,
                titlefont: {
                  size: chartStyling.size,
                  color: chartStyling.textColor,
                },
                tickfont: {
                  size: chartStyling.size,
                  color: chartStyling.textColor,
                },

                plot_bgcolor: chartStyling.bgColor,
                paper_bgcolor: chartStyling.bgColor,
                border_radius: chartStyling.borderRadius,
                displayModeBar: false,
                margin: {
                  l: chartStyling.marginLeft,
                  t: chartStyling.marginTop,
                  r: chartStyling.marginRight,
                  b: chartStyling.marginBottom,
                },
                width: 800,
                height: 350,
                xaxis: {
                  automargin: true,
                  showgrid: true,
                  gridcolor: chartStyling.gridcolor,
                  mirror: chartStyling.mirrorTicks,
                  showline: true,
                  linecolor: chartStyling.lcolor,
                  rangeslider: {},
                  rangeselector: Labels[i],
                  tickfont: {
                    size: chartStyling.size,
                    color: chartStyling.textColor,
                  },
                  title: {
                    text: Labels[i]["xLabel"],
                    font: {
                      size: chartStyling.size,
                      color: chartStyling.textColor,
                    },
                  },
                },
                yaxis: {
                  showgrid: true,
                  automargin: true,
                  gridcolor: chartStyling.gridcolor,
                  mirror: chartStyling.mirrorTicks,
                  showline: true,
                  linecolor: chartStyling.lcolor,
                  fixedrange: true,
                  tickfont: {
                    size: chartStyling.size,
                    color: chartStyling.textColor,
                  },

                  title: {
                    text: Labels[i]["yLabel"],
                    font: {
                      size: chartStyling.size,
                      color: chartStyling.textColor,
                    },
                  },
                },
              }}
              config={{
                responsive: true,
                useResizeHandler: true,
                displaylogo: false,
                displayModeBar: true,
              }}
            />
          </div>
        ))}
    </div>
  );
}

export default Chart;
