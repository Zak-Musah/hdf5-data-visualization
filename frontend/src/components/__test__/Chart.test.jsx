import React from "react";
import { render } from "enzyme";
import renderer from "react-test-renderer";

import Chart from "../Chart/Chart";

test("Chart renders properly", () => {
  const wrapper = render(<Chart />);
  expect(wrapper.find(".useEffect"));
});

test("Chart renders with snapshot", () => {
  const tree = renderer.create(<Chart />).toJSON();
  expect(tree).toMatchSnapshot();
});
