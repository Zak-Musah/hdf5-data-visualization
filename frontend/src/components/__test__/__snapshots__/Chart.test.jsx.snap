// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart renders with snapshot 1`] = `
<div
  className="row"
/>
`;
